packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -59,6 +59,20 @@ const requestPromiseWithDefaults = requestPromise.defaults({
 	timeout: 300000, // 5 minutes
 });
 
+/**
+ * Returns binary data buffer for given item index and property name.
+ *
+ * @export
+ * @param {ITaskDataConnections} inputData
+ * @param {number} itemIndex
+ * @param {string} propertyName
+ * @returns {Promise<Buffer>}
+ */
+export async function getBinaryDataBuffer(inputData: ITaskDataConnections, itemIndex: number, propertyName: string): Promise<Buffer> {
+	const binaryData = inputData['main']![0]![itemIndex]!.binary![propertyName]!;
```

Comment:
I am in doubt about this `0` index here. It is related to the first execution of a node, but perhaps if we've built a loop, won't it always get the files from the first iteration? And worse, if we have multiple executions and the number of files vary, this can even cause some `index out of bounds` errors, wouldn't it?

Reply:
Great find. Did fix it.

Reply:
Actually worked fine for multiple runs but added now support for multiple inputs.